File: src/security/plan-hasher.ts
import crypto from 'crypto';

export class PlanHasher {
  static computeHash(actionPlan: any): string {
    const serialized = JSON.stringify(actionPlan, Object.keys(actionPlan).sort());
    return crypto.createHash('sha256').update(serialized).digest('hex');
  }

  static generateNonce(): string {
    return crypto.randomBytes(16).toString('hex');
  }
}

File: src/security/plan_hasher.py
import hashlib
import json
import secrets
from typing import Any


class PlanHasher:
    @staticmethod
    def compute_hash(action_plan: Any) -> str:
        serialized = json.dumps(action_plan, sort_keys=True)
        return hashlib.sha256(serialized.encode("utf-8")).hexdigest()

    @staticmethod
    def generate_nonce() -> str:
        return secrets.token_hex(16)
